### PR TITLE
CB-1174 Add environments to database servers, and more

### PR DESCRIPTION
**Pull request by bhavanki**
A database server entity in redbeams is now associated with one or more
environment IDs. The list API is connected up properly so that the
environment ID query parameter matters.

Other useful changes:

- The query for finding all database server entities is expanded to
include the environment ID as a criterion. The query is not fully tested
yet to see what happens if no environment ID is specified.
- The database vendor property for database server API models is
refactored into the base model class.
- The databaseserverconfig table gets a new constraint pertaining to
archiving.
- Many commented-out lines, left tentatively in place, are cleaned up.
(There are still plenty more in redbeams to dispose of later.)

### CODE REVIEW

File: redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/base/DatabaseServerV4Base.java [hunk 1]
```diff
@@ -38,11 +38,15 @@
...
+    @ApiModelProperty(value = DatabaseServer.DATABASE_VENDOR, required = true)
+    private String databaseVendor;
+
     @ApiModelProperty(DatabaseServer.CONNECTOR_JAR_URL)
     private String connectorJarUrl;
 
-    // @ApiModelProperty(ModelDescriptions.ENVIRONMENTS)
-    // private Set<String> environments = new HashSet<>();
+    @ApiModelProperty(ModelDescriptions.ENVIRONMENT_IDS)
+    private Set<Long> environmentIds = new HashSet<>();
```
Comment (gergopapi2) [inline redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/base/DatabaseServerV4Base.java]: 1. Current info: 1 rds can be part of 1 env only.
2. We will have to store the CRN of the env

File: redbeams/src/main/resources/schema/app/20190508131601_CB-1174_database_server_config_env.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+-- CB-1174 database server config environment IDs
```
Comment (gergopapi2) [inline redbeams/src/main/resources/schema/app/20190508131601_CB-1174_database_server_config_env.sql]: We should store a CRN for every databaseServer as well. That will be the id to return.

File: redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/DatabaseServerV4Endpoint.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = DatabaseServerOpDescription.LIST_BY_WORKSPACE, produces = ContentType.JSON, notes = Notes.DATABASE_SERVER_NOTES,
         nickname = "listDatabasesServersByWorkspace")
-    DatabaseServerV4Responses list(@PathParam("workspaceId") Long workspaceId, @QueryParam("environment") String environment,
+    DatabaseServerV4Responses list(@PathParam("workspaceId") Long workspaceId, @QueryParam("environmentId") Long environmentIds,
```
Comment (gergopapi2) [inline redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/DatabaseServerV4Endpoint.java]: environment id is a CRN - this change, however, can be part of another PR if you think it is better to do things more fine-grained.

File: redbeams/src/main/java/com/sequenceiq/redbeams/repository/DatabaseServerConfigRepository.java [hunk 4]
```diff
@@ -15,6 +16,9 @@
 @Transactional(TxType.REQUIRED)
 public interface DatabaseServerConfigRepository extends CrudRepository<DatabaseServerConfig, Long> {
 
-    Set<DatabaseServerConfig> findAllByWorkspaceId(Long workspaceId);
+    @Query("SELECT s FROM DatabaseServerConfig s WHERE "
+            + "s.workspaceId = :workspaceId AND :environmentId MEMBER OF s.environmentIds AND "
+            + "s.resourceStatus = 'USER_MANAGED'")
```
[the remark this answers is not in the document]
Comment (bhavanki) [reply redbeams/src/main/java/com/sequenceiq/redbeams/repository/DatabaseServerConfigRepository.java]: I think so. Here, it seems useful to track whether a database server was registered as already created (to me, that's USER_MANAGED) or was created by redbeams itself (some other status, maybe not DEFAULT).

File: redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/base/DatabaseServerV4Base.java (hunk 1 above)
Comment (bhavanki) [reply redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/base/DatabaseServerV4Base.java]: OK - changing this to a simple String column then.

File: redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/DatabaseServerV4Endpoint.java (hunk 3 above)
Comment (bhavanki) [reply redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/DatabaseServerV4Endpoint.java]: I can definitely update it now.

File: redbeams/src/main/resources/schema/app/20190508131601_CB-1174_database_server_config_env.sql (hunk 2 above)
Comment (bhavanki) [reply redbeams/src/main/resources/schema/app/20190508131601_CB-1174_database_server_config_env.sql]: I will punt on this one for now, because I don't know yet how redbeams will determine the CRN for databases and database servers.
